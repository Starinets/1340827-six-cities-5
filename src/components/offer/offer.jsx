import React from 'react';
import * as Type from '../../types';
import {NavLink} from "react-router-dom";

import PremiumLabel from '../premium-label/premium-label';
import OfferImageList from '../offer-image-list/offer-image-list';
import OfferFeatureList from '../offer-feature-list/offer-feature-list';
import OfferHost from '../offer-host/offer-host';
import OfferReviewList from '../offer-review-list/offer-review-list';
import OfferCard from '../offer-card/offer-card';
import ReviewForm from '../review-form/review-form';
import Map from '../map/map';

import {transformRatingToWidth} from '../../utils';
import {
  MapPlace,
  OfferPlace
} from '../../constants';

const Offer = ({offerID, offers}) => {

  const SINGULAR_MEANING = 2;

  const offer = offers[offerID];

  if (offer === undefined) {
    return (null);
  }

  const width = transformRatingToWidth(offer.rating);

  return (
    <div className="page">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <NavLink className="header__logo-link" to="/">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
              </NavLink>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <NavLink className="header__nav-link header__nav-link--profile" to="/">
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__login">Sign in</span>
                  </NavLink>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              <OfferImageList imageList={ offer.images } />
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              <PremiumLabel isPremium={ offer.isPremium } />
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  { offer.name }
                </h1>
                <button className={ `property__bookmark-button ${ offer.isFavorite ? `property__bookmark-button--active` : `` } button` } type="button">
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={ {width} }></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{ offer.rating }</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  { offer.type }
                </li>
                <li className="property__feature property__feature--bedrooms">
                  { offer.bedroomsCount < SINGULAR_MEANING ? `${ offer.bedroomsCount } Bedroom` : `${ offer.bedroomsCount } Bedrooms`}
                </li>
                <li className="property__feature property__feature--adults">
                  { offer.adult < SINGULAR_MEANING ? `Max ${ offer.adultsCount } adult` : `Max ${ offer.adultsCount } adults`}
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{ offer.price }</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  <OfferFeatureList featureList={ offer.features }/>
                </ul>
              </div>
              <OfferHost offer={ offer } />
              <section className="property__reviews reviews">
                <OfferReviewList reviewList={ offer.reviews } />
                <ReviewForm />
              </section>
            </div>
          </div>
          <Map
            offers = { offers }
            mapPlace = { MapPlace.OFFER }
          />
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              {
                offers.map((neighbor, index) => {
                  return (
                    <OfferCard
                      key = { index }
                      offer = { neighbor }
                      offerPlace = { OfferPlace.NEIGHBORHOOD }
                    />
                  );
                })
              }
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

Offer.propTypes = {
  offerID: Type.STRING,
  offers: Type.OFFERS,
};

export default Offer;
