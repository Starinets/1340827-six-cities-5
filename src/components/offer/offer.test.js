import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {Offer} from './offer';
import {
  emptyFunction,
  offer,
  Offers,
  reviews,
  matchOfferID
} from '../../test-data/test-data';

it(`Component rendered correctly -> Offer`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <Offer
        offer={ offer }
        reviews={ reviews }
        neighborhoods={ Offers }
        getCurrentOffer={ emptyFunction }
        removeActiveCard={ emptyFunction }
        match={ matchOfferID }
      />
  );
  expect(tree).toMatchSnapshot();
});
