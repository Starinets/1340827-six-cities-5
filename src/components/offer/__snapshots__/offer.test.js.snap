// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendered correctly -> Offer 1`] = `
<div
  className="page"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <OfferImageList
            imageList={
              Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ]
            }
          />
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <PremiumLabel
            isPremium={false}
          />
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              The house among olive 
            </h1>
            <button
              className="property__bookmark-button  button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
            <Memo(Connect(FavoriteButton))
              buttonHeight={33}
              buttonWidth={31}
              className="property"
              id={1}
              isFavorite={false}
            />
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3.8
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              house
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3 Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 4 adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              493
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <OfferFeatureList
                featureList={
                  Array [
                    "Laptop friendly workspace",
                    "Washer",
                    "Towels",
                    "Fridge",
                    "Air conditioning",
                    "Baby seat",
                    "Breakfast",
                  ]
                }
              />
            </ul>
          </div>
          <OfferHost
            offer={
              Object {
                "adultsCount": 4,
                "bedroomsCount": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 50.938361,
                    "longitude": 6.959974,
                    "zoom": 13,
                  },
                  "name": "Cologne",
                },
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "features": Array [
                  "Laptop friendly workspace",
                  "Washer",
                  "Towels",
                  "Fridge",
                  "Air conditioning",
                  "Baby seat",
                  "Breakfast",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 50.957361,
                  "longitude": 6.9509739999999995,
                  "zoom": 16,
                },
                "name": "The house among olive ",
                "price": 493,
                "rating": 3.8,
                "type": "house",
              }
            }
          />
          <section
            className="property__reviews reviews"
          >
            <OfferReviewList
              reviewList={
                Array [
                  Object {
                    "date": "2020-11-01T13:38:44.752Z",
                    "id": 1,
                    "rating": 5,
                    "text": "The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.",
                    "user": Object {
                      "avatar": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg",
                      "id": 14,
                      "isPro": true,
                      "name": "Corey",
                    },
                  },
                  Object {
                    "date": "2020-11-01T13:38:44.752Z",
                    "id": 2,
                    "rating": 3,
                    "text": "The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.",
                    "user": Object {
                      "avatar": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg",
                      "id": 12,
                      "isPro": true,
                      "name": "Isaac",
                    },
                  },
                  Object {
                    "date": "2020-11-01T13:38:44.752Z",
                    "id": 3,
                    "rating": 3,
                    "text": "I stayed here for one night and it was an unpleasant experience.",
                    "user": Object {
                      "avatar": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg",
                      "id": 15,
                      "isPro": false,
                      "name": "Kendall",
                    },
                  },
                ]
              }
            />
            <Memo(Connect(WithReviewForm))
              id={1}
            />
          </section>
        </div>
      </div>
      <Memo(Connect(Map))
        mapPlace="property"
        offers={
          Array [
            Object {
              "adultsCount": 4,
              "bedroomsCount": 3,
              "city": Object {
                "location": Object {
                  "latitude": 50.938361,
                  "longitude": 6.959974,
                  "zoom": 13,
                },
                "name": "Cologne",
              },
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "features": Array [
                "Laptop friendly workspace",
                "Washer",
                "Towels",
                "Fridge",
                "Air conditioning",
                "Baby seat",
                "Breakfast",
              ],
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 50.957361,
                "longitude": 6.9509739999999995,
                "zoom": 16,
              },
              "name": "The house among olive ",
              "price": 493,
              "rating": 3.8,
              "type": "house",
            },
            Object {
              "adultsCount": 8,
              "bedroomsCount": 3,
              "city": Object {
                "location": Object {
                  "latitude": 50.846557,
                  "longitude": 4.351697,
                  "zoom": 13,
                },
                "name": "Brussels",
              },
              "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
              "features": Array [
                "Towels",
                "Baby seat",
                "Laptop friendly workspace",
                "Dishwasher",
                "Washing machine",
                "Air conditioning",
                "Washer",
                "Coffee machine",
                "Breakfast",
                "Fridge",
              ],
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 2,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 50.839557,
                "longitude": 4.346697,
                "zoom": 16,
              },
              "name": "Amazing and Extremely Central Flat",
              "price": 323,
              "rating": 4.3,
              "type": "apartment",
            },
            Object {
              "adultsCount": 7,
              "bedroomsCount": 4,
              "city": Object {
                "location": Object {
                  "latitude": 50.846557,
                  "longitude": 4.351697,
                  "zoom": 13,
                },
                "name": "Brussels",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "features": Array [
                "Washer",
                "Laptop friendly workspace",
                "Breakfast",
                "Air conditioning",
                "Baby seat",
              ],
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 50.865556999999995,
                "longitude": 4.371696999999999,
                "zoom": 16,
              },
              "name": "The house among olive ",
              "price": 218,
              "rating": 4.6,
              "type": "apartment",
            },
            Object {
              "adultsCount": 7,
              "bedroomsCount": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.897976,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "features": Array [
                "Dishwasher",
                "Air conditioning",
                "Laptop friendly workspace",
                "Washer",
                "Baby seat",
                "Towels",
                "Breakfast",
                "Coffee machine",
                "Fridge",
              ],
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 4,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.385540000000006,
                "longitude": 4.902976,
                "zoom": 16,
              },
              "name": "Perfectly located Castro",
              "price": 440,
              "rating": 4.6,
              "type": "house",
            },
          ]
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighborhood
        </h2>
        <Memo(Connect(WithOfferList))
          offerPlace="near-places__"
          offers={
            Array [
              Object {
                "adultsCount": 4,
                "bedroomsCount": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 50.938361,
                    "longitude": 6.959974,
                    "zoom": 13,
                  },
                  "name": "Cologne",
                },
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "features": Array [
                  "Laptop friendly workspace",
                  "Washer",
                  "Towels",
                  "Fridge",
                  "Air conditioning",
                  "Baby seat",
                  "Breakfast",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 50.957361,
                  "longitude": 6.9509739999999995,
                  "zoom": 16,
                },
                "name": "The house among olive ",
                "price": 493,
                "rating": 3.8,
                "type": "house",
              },
              Object {
                "adultsCount": 8,
                "bedroomsCount": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 50.846557,
                    "longitude": 4.351697,
                    "zoom": 13,
                  },
                  "name": "Brussels",
                },
                "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                "features": Array [
                  "Towels",
                  "Baby seat",
                  "Laptop friendly workspace",
                  "Dishwasher",
                  "Washing machine",
                  "Air conditioning",
                  "Washer",
                  "Coffee machine",
                  "Breakfast",
                  "Fridge",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 50.839557,
                  "longitude": 4.346697,
                  "zoom": 16,
                },
                "name": "Amazing and Extremely Central Flat",
                "price": 323,
                "rating": 4.3,
                "type": "apartment",
              },
              Object {
                "adultsCount": 7,
                "bedroomsCount": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 50.846557,
                    "longitude": 4.351697,
                    "zoom": 13,
                  },
                  "name": "Brussels",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "features": Array [
                  "Washer",
                  "Laptop friendly workspace",
                  "Breakfast",
                  "Air conditioning",
                  "Baby seat",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 3,
                "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 50.865556999999995,
                  "longitude": 4.371696999999999,
                  "zoom": 16,
                },
                "name": "The house among olive ",
                "price": 218,
                "rating": 4.6,
                "type": "apartment",
              },
              Object {
                "adultsCount": 7,
                "bedroomsCount": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "features": Array [
                  "Dishwasher",
                  "Air conditioning",
                  "Laptop friendly workspace",
                  "Washer",
                  "Baby seat",
                  "Towels",
                  "Breakfast",
                  "Coffee machine",
                  "Fridge",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 4,
                "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.385540000000006,
                  "longitude": 4.902976,
                  "zoom": 16,
                },
                "name": "Perfectly located Castro",
                "price": 440,
                "rating": 4.6,
                "type": "house",
              },
            ]
          }
        />
      </section>
    </div>
  </main>
</div>
`;
